cmd/shp/main.go

```diff
@@ -51,4 +67,10 @@ func initPFlags() {
 	flags := pflag.NewFlagSet(ApplicationName, pflag.ExitOnError)
 	flags.AddGoFlagSet(goflag.CommandLine)
 	pflag.CommandLine = flags
+
+	for _, flag := range hiddenLogFlags {
+		if err := flags.MarkHidden(flag); err != nil {
```

Comment:
Interesting approach. Early on I was trying to achieve the same but using [ConfigOverrideFlags](https://github.com/kubernetes/client-go/blob/af4295f5013b55c45b05f1e7850f7fa9c4dbfdb4/tools/clientcmd/overrides.go#L40-L48), and it does take more effort to understand and adapt the existing bits.

Reply:
HI @otaviof, thank you for the review. I have to admit that I never implemented a CLI in Go and therefore was in experimenting mode when I put together this PR. Are you suggesting that I should use the constants from https://github.com/kubernetes/client-go/blob/af4295f5013b55c45b05f1e7850f7fa9c4dbfdb4/tools/clientcmd/overrides.go#L146-L164 instead of having the flag names hard-coded in our code?

Reply:
Yes, I'm saying we could alternatively follow that path. Although, this implementation is a straightforward way leveraging Cobra directly, I think it's a good way to start.